google/cloud/aiplatform/utils/featurestore_utils.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+from typing import Dict
+
+RESOURCE_ID_PATTERN_REGEX = r"[a-z_][a-z0-9_]{0,59}"
+FEATURESTORE_NAME_PATTERN_REGEX = (
+    r"projects\/(?P<project>[\w-]+)"
+    r"\/locations\/(?P<location>[\w-]+)"
+    r"\/featurestores\/(?P<featurestore_id>" + RESOURCE_ID_PATTERN_REGEX + r")"
+)
+ENTITY_TYPE_NAME_PATTERN_REGEX = (
+    FEATURESTORE_NAME_PATTERN_REGEX
+    + r"\/entityTypes\/(?P<entity_type_id>"
+    + RESOURCE_ID_PATTERN_REGEX
+    + r")"
+)
+FEATURE_NAME_PATTERN_REGEX = (
+    ENTITY_TYPE_NAME_PATTERN_REGEX
+    + r"\/features\/(?P<feature_id>"
+    + RESOURCE_ID_PATTERN_REGEX
```

Comment:
Can you add more check for feature_id -- it cannot be one of reserved words: entity_id, feature_timestamp, arrival_timestamp.

Reply:
Vertex SDK is relying on the service for this type of value validation, most validation we are doing is to assist certain usage within the SDK which is to prepare the value to be passed into the service, i.e. these pattern validations essentially are to assist the limitation due to lack of overloading constructor, and we are validating the arguments and form the final resource name to pass into the service. 

However, if the service does not throw an error on the validation you proposed and will cause un alerted error, we can definitely add these in the SDK layer.

Please let me know!